connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginUtils.java

```diff
@@ -386,6 +362,25 @@ private static String prunePluginName(PluginDesc<?> plugin, String suffix) {
         return simple;
     }
 
+    public static Map<String, String> computeAliases(PluginScanResult scanResult) {
+        Map<String, Set<String>> aliasCollisions = new HashMap<>();
+        scanResult.forEach(pluginDesc -> {
+            aliasCollisions.computeIfAbsent(simpleName(pluginDesc), ignored -> new HashSet<>()).add(pluginDesc.className());
+            aliasCollisions.computeIfAbsent(prunedName(pluginDesc), ignored -> new HashSet<>()).add(pluginDesc.className());
+        });
+        Map<String, String> aliases = new HashMap<>();
+        for (Map.Entry<String, Set<String>> entry : aliasCollisions.entrySet()) {
+            String alias = entry.getKey();
+            Set<String> classNames = entry.getValue();
+            if (classNames.size() == 1) {
+                aliases.put(alias, classNames.stream().findAny().get());
+            } else {
+                log.warn("Ambiguous alias '{}' refers to multiple distinct plugins {}: ignoring", alias, classNames);
```

Comment:
After seeing an example of this log message, I'm wondering if the trailing `: ignoring` part here is really beneficial. Maybe something like this would be a bit clearer?

```suggestion
                log.warn("Ignoring ambiguous alias '{}' since it refers to multiple distinct plugins {}", alias, classNames);
```